src/engine/strat_engine/backstore/crypt.rs

```diff
@@ -221,86 +237,159 @@ impl CryptInitializer {
             MetadataSize::try_from(DEFAULT_CRYPT_METADATA_SIZE)?,
             KeyslotsSize::try_from(DEFAULT_CRYPT_KEYSLOTS_SIZE)?,
         )?;
-        let result = self.initialize_with_err(&mut device, key_description);
-        match result {
-            Ok(activated_path) => Ok(CryptHandle::new(
-                self.physical_path,
-                activated_path,
-                self.identifiers,
-                EncryptionInfo {
-                    key_description: key_description.clone(),
-                    clevis_info: None,
-                },
-                self.activation_name,
-            )),
+        let result = self.initialize_with_err(device, key_description, clevis_parsed);
+        let mut device = match self.acquire_crypt_device() {
+            Ok(d) => d,
             Err(e) => {
-                if let Err(err) = Self::rollback(device, &self.physical_path, self.activation_name)
+                warn!(
+                    "Failed to roll back crypt device initialization; you \
+                    may need to manually wipe this device: {}",
+                    e,
+                );
+                return Err(e);
+            }
+        };
+
+        result
+            .and_then(|activated_path| {
+                Ok(CryptHandle::new(
+                    self.physical_path.clone(),
+                    activated_path,
+                    self.identifiers,
+                    EncryptionInfo {
+                        key_description: key_description.cloned(),
+                        clevis_info: clevis_info_from_metadata(&mut device)?,
+                    },
+                    self.activation_name.clone(),
+                ))
+            })
+            .map_err(|e| {
+                if let Err(err) =
+                    Self::rollback(&mut device, &self.physical_path, self.activation_name)
                 {
                     warn!(
                         "Failed to roll back crypt device initialization; you may need \
-                        to manually wipe this device: {}",
+                    to manually wipe this device: {}",
```

Comment:
Indentation may be weird here.

Reply:
How do you want to approach this moving forward? rustfmt doesn't do a very good job of handling this so would you like me to keep it as one long string? We can't rely on it correctly putting in the indentation.

Reply:
Yes. rustfmt is basically hopeless w/ text inside macros. I've given up and just use the one long string. We can go with this for as long as rustfmt can't do any better.